gridded_etl_tools/utils/store.py

```diff
@@ -43,19 +43,18 @@ def __init__(self, dm: dataset_manager.DatasetManager):
         self.dm = dm
 
     @abstractmethod
-    def mapper(self, **kwargs: dict) -> collections.abc.MutableMapping:
+    def mapper(self, **kwargs) -> collections.abc.MutableMapping:
         """
         Parameters
         ----------
         **kwargs : dict
-            Implementation specific keywords
+            Implementation specific keywords. TODO: standardize interface across implementations
```

Comment:
I understand inherited function signatures need to be compatible with the parent to satisfy liskov substitution but my understanding is using kwargs as a way to satisfy it is acceptable practice in Python? I'm thinking of something like this for example https://stackoverflow.com/a/54155637

Reply:
That works, but I would discourage it, because the point of having an interface is to provide polymorphism, which means that calling code can/should be written in a way that is unaware of which implementation of an interface it is using. In order to access those extra keyword arguments, the caller has to be aware that it has a more specific implementation than the generic interface. It's best to avoid doing that if you can. 

In this particular instance, I could imagine bringing `refresh` up to interface level, since I can imagine that being appropriate universally. I can also imagine moving `set_root` from `mapper` to the constructor for `IPLD`, since the constructor is not part of `StoreInterface` and doesn't need to be generic, and we can probably make that decision when instantiating the store.

Reply:
Yeah that design seems better but I guess there will still be different return types for mapper so for example you can call set_root on an IPLD mapper but not an S3 mapper

Reply:
Ideally the `Mapper` interface would get standardized as well, but one step at a time, and we don't always achieve ideal. (We mostly don't.) The more we keep things like this in mind, though, the better designs we'll make when writing new code.